cdap-ui/app/cdap/components/PipelineDetails/PipelineButtonsWrapper/index.js

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+require('./PipelineButtonsWrapper.scss');
+
+export default class PipelineButtonsWrapper extends Component {
+  state = {
+    activeButtons: []
```

Comment:
what are the cases where there are multiple active buttons?

Reply:
So before, I tried having just a single value to indicate which button is active. On hovering over a button, I set that button as active, and on hovering out, set the active button to null. 

However, there was an issue when I moved the mouse quickly from one button to its neighbor. Sometimes the onMouseEnter callback of the 2nd button would trigger first, setting that button as active, then the onMouseLeave callback of the 1st one triggers, setting the active button to null. 

I wasn't sure how to get around this issue, so I changed this to be an array, so the active button value wouldn't get overwritten.